analyzers/src/SonarAnalyzer.CSharp/Rules/ClassShouldNotBeEmpty.cs

```diff
@@ -51,7 +51,17 @@ public sealed class ClassShouldNotBeEmpty : ClassShouldNotBeEmptyBase<SyntaxKind
             SyntaxKind.ElseDirectiveTrivia,
             SyntaxKind.EndIfDirectiveTrivia));
 
-    private bool IsParameterlessRecord(SyntaxNode node) =>
+    private static bool IsParameterlessRecord(SyntaxNode node) =>
+        RecordDeclarationSyntax(node) is { ParameterList.Parameters.Count: 0 } declaration
+        && !ProvidesBaseWithParameters(declaration);
+
+    private static bool ProvidesBaseWithParameters(RecordDeclarationSyntaxWrapper node) =>
+        node.BaseList?.Types.FirstOrDefault() is { } type
+        && PrimaryConstructorBaseTypeSyntaxWrapper.IsInstance(type)
+        && (PrimaryConstructorBaseTypeSyntaxWrapper)type is { ArgumentList.Arguments.Count: >= 1 };
+
+    private static RecordDeclarationSyntaxWrapper? RecordDeclarationSyntax(SyntaxNode node) =>
```

Comment:
This method isn't really needed anymore. All it does is check if the node is a record declaration or not. Moving that to `IsParameterlessRecord` will make the code less cluttered.

Reply:
I've moved some stuff around. I think it improves the readability.